pass.go

```diff
@@ -0,0 +1,1054 @@
+}
+
+func preparseTemplates(p *Pass) {
+	p.sv.templates = parsedTemplates{
+		CreateGet:  p.LoadTemplate("create/get.html"),
+		CreatePost: p.LoadTemplate("create/post.html"),
+
+		GetGet:  p.LoadTemplate("get/get.html"),
+		GetPost: p.LoadTemplate("get/post.html"),
+
+		Errors400:        p.LoadTemplate("errors/400.html"),
+		Errors404:        p.LoadTemplate("errors/404.html"),
+		Errors405:        p.LoadTemplate("errors/405.html"),
+		Errors500:        p.LoadTemplate("errors/500.html"),
+		ErrorsPwNotFound: p.LoadTemplate("errors/pw_not_found.html"),
+	}
+}
+
+// loadConfig loads the config file from the given filename
+func loadConfig(configFileName string) (cfg passConfiguration, err error) {
+	var file *os.File
+	if file, err = os.Open(configFileName); err != nil {
+		return cfg, fmt.Errorf("Config file (%s) was not found, please create one using the included template", configFileName)
+	}
+
+	fmt.Printf("Loading config file \"%s\"\n", configFileName)
+
+	decoder := json.NewDecoder(file)
+	if err = decoder.Decode(&cfg); err != nil {
+		return cfg, fmt.Errorf("Unable to decode config file: %v", err)
+	}
+
+	if len(cfg.Secret) < 64 {
+		return cfg, fmt.Errorf("Site secret shorter than 64 characters, please make it at least 64 characters")
+	}
+
+	fmt.Printf("Config file \"%s\" successfully loaded\n----\n", configFileName)
+	return cfg, nil
+}
+
+// recoverWrap registers a deferred error handler for the http.Handler
+// This prevents the entire program from coming down like a ton of bricks if a
+// user causes a panic.
+// The stacktrace is printed and the corresponding handler dies, but no other
+// requests are affected.
+func recoverWrap(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		defer func() {
+			r := recover()
+			if r != nil {
+				switch t := r.(type) {
+				case string:
+					err = errors.New(t)
+				case error:
+					err = t
+				default:
+					err = fmt.Errorf("recoverWrap Unknown error %T %v", r, r)
+				}
+
+				log.Printf("recoverWrap triggered!\n")
+				fmt.Printf("Cause: %v\n|\n", err)
+
+				msgStrings := strings.Split(strings.TrimRight(string(debug.Stack()), "\n"), "\n")
+
+				// 7 is to remove the heading and the first 3 frames (2 lines each)
+				// because they contain the call to the recoverWrap func and the
+				// call to panic() which is useless when _this_ program is the issue
+				msgStrings = msgStrings[7:]
+
+				for _, value := range msgStrings {
+					fmt.Printf("|\t%s\n", string(value))
+				}
+
+				fmt.Printf("|\nEND of wrapped panic\n")
+
+				// We paniced, assume templating is a no-go and just output plaintext error
+				http.Error(w, "Server error", http.StatusInternalServerError)
+			}
+		}()
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+// RequestHandler is the interface an importer must implement to hook into the
+// request handler.
+type RequestHandler interface {
+	Handle(hi http.Handler) (ho http.Handler)
```

Comment:
`hi` and `ho` sounds like Santa Claus.

Reply:
`RequestHandler` interface was unused, removed.